templates/object-cache.php

```
@@ -327,65 +327,130 @@ private function connect_redis() {
-					if ( defined( '\Redis::SERIALIZER_IGBINARY' ) ) {
-						$this->redis->setOption( \Redis::OPT_SERIALIZER, \Redis::SERIALIZER_IGBINARY ); // Optimize serialization if igbinary is available..
-					} else {
-						$this->redis->setOption( \Redis::OPT_SERIALIZER, \Redis::SERIALIZER_PHP );
+				if ( 'cluster' === $mode && class_exists( 'RedisCluster' ) ) {
+					$nodes = isset( $config['nodes'] ) ? (array) $config['nodes'] : array();
+					if ( $use_tls ) {
+						$nodes = array_map( function( $node ) {
+							return ( strpos( $node, 'tls://' ) === 0 ) ? $node : 'tls://' . $node;
+						}, $nodes );
 					}
+					$this->redis           = new \RedisCluster( null, $nodes, $timeout, $timeout, true, $password );
```

PHPStan (2.1.46)

[error] Property WP_Object_Cache::$redis (Redis) does not accept RedisCluster.


(assign.propertyType)